pkg/oauth/registry/helpers/converter.go

```diff
@@ -0,0 +1,140 @@
+package helpers
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	kubeerr "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+type ObjectDecoratorFunc func(obj runtime.Object) runtime.Object
+type ObjectFilterFunc func(ctx api.Context, obj runtime.Object) error
+
+type ListDecoratorFunc func(obj runtime.Object) runtime.Object
+type ListFilterFunc func(ctx api.Context, options *api.ListOptions) error
+
+type ObjectNameMutatorFunc func(ctx api.Context, name string) (string, error)
+
+type filterConverter struct {
+	storage  ReadAndDeleteStorage
+	objDec   ObjectDecoratorFunc
+	objFil   ObjectFilterFunc
+	listDec  ListDecoratorFunc
+	listFil  ListFilterFunc
+	namer    ObjectNameMutatorFunc
+	resource unversioned.GroupResource
+}
+
+type ReadAndDeleteStorage interface {
+	rest.Storage
+	rest.Getter
+	rest.Lister
+	rest.GracefulDeleter
+	rest.CollectionDeleter
+	rest.Watcher
+}
+
+var _ ReadAndDeleteStorage = &filterConverter{}
+
+func NewFilterConverter(
+	storage ReadAndDeleteStorage,
+	objDec ObjectDecoratorFunc,
```

Comment:
Name these parameters exactly like the type, it'll more readable. Generally, I'd drop func types, but name the parameters better. You read parameter names when going through the code when implementing them rather then types.